scripts/operations/system_recovery/bos_cli.py

```diff
@@ -0,0 +1,110 @@
+#
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Module to provide BOS CLI functions
+"""
+
+import json
+import subprocess
+import tempfile
+
+from typing import List
+
+from bos_session_templates import BosError, BosSessionTemplate
+
+class BosCliError(BosError):
+    pass
+
+def create_session_template(session_template: BosSessionTemplate, bos_version: int) -> None:
+    """
+    Wrapper for calling the CLI to create the specified BOS session template using the
+    specified BOS version.
+    Returns nothing.
+    Raises BosCliError on error.
+    """
+    template_name = session_template["name"]
+    template_to_create = session_template.copy()
+    # Delete the name field (since it is specified on the command line of the create command)
+    del template_to_create["name"]
+
+    # Write template to a temporary file and create the record
+    with tempfile.NamedTemporaryFile(mode="wt", suffix=".json", prefix="session_template") as tmp:
+        json.dump(template_to_create, tmp)
+        # Make sure the data has been written to the file so the CLI command can read it.
+        tmp.flush()
+        if bos_version == 1:
+            create_command = ["cray", "bos", f"v{bos_version}", "sessiontemplate", "create",
+                              "--file", tmp.name, "--name", template_name, "--format", "json"]
+        else:
+            create_command = ["cray", "bos", f"v{bos_version}", "sessiontemplates", "create",
+                              "--file", tmp.name, template_name, "--format", "json"]
+        try:
+            subprocess.run(create_command, stdout=subprocess.PIPE, check=True)
```

Comment:
Is there a reason to be using the cli rather than api calls here?  Typically we recommend against that for scripts.

Reply:
I only did this because it's what the existing BOS and IMS import/export scripts did. I agree with you that using the API makes more sense. For now, since I was already making a lot of other changes, I didn't want to make this change on top of that. But at some point I meant to follow up and find out who on the team authored the existing scripts and see why they chose to use the CLI instead of the API. Since it was done for both IMS and BOS, it seemed unlikely to be a coincidence, given that different people on the team owned those.